iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsHandlingCheck.java

```diff
@@ -55,15 +59,16 @@ public class EnvSecretCheck implements IacCheck {
   private static final Set<String> EXCLUSIONS = Set.of("ALLOW", "DIR", "EXPIRE", "EXPIRY", "FILE", "ID",
     "LOCATION", "NAME", "OWNER", "PATH", "URL");
 
+  // Patterns to split the identifier of a variable into separate words
   private static final Pattern UNDERSCORE_NAME_PATTERN = Pattern.compile("^\\w+$");
   private static final Pattern DASH_NAME_PATTERN = Pattern.compile("^[a-zA-Z0-9-]+$");
-
   private static final Pattern CAMELCASE_NAME_PATTERN = Pattern.compile("^[A-Za-z]+$");
-
   private static final Pattern CAMELCASE_SPLIT_PATTERN = Pattern.compile("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])");
 
+  // Pattern to identify a URL
   private static final Pattern URL_PATTERN = Pattern.compile("^(http|ftp)s?://");
```

Comment:
Do you think this is intended by the rule?

Reply:
To my understanding we want to check if the value is a secret or not, by excluding some known pattern.
One of them is detecting if it looks like an URL.

I like the conservative approach that "we don't know which protocol can be used, but potentially any" and not just http/ftp, even if the probably represent 90%+ use case. Since I'm not proficient with all existing protocols, I prefer to see larger, for example in my previous company some tools were using sftp and not ftps, maybe also an sftp url can exist ? Dunno :D
Also there is still a risk that we exclude a real secret token matching this more general regex.

I don't have a strong opinion on this, we can also go the current way : exclude http/ftp as it should cover most cases, and if eventually later a user report a FP, we can extend or choose another approach.